cmd/go-yaml/parser.go

```diff
@@ -29,10 +29,9 @@ func (p *Parser) Next() (*Token, error) {
 	}
 
 	var yamlToken libyaml.Token
-	if !p.parser.Scan(&yamlToken) {
-		if p.parser.ErrorType != libyaml.NO_ERROR {
-			return nil,
-				fmt.Errorf("parser error: %v", p.parser.Problem)
+	if err := p.parser.Scan(&yamlToken); err != nil {
+		if err != io.EOF {
```

Comment:
Please use

```suggestion
		if !errors.Is(err, io.EOF) {
```
It would be safer if something wraps the error later